src/tests/ftest/launch.py

```diff
@@ -52,2294 +68,2973 @@
-                "Not creating stacktrace".format(gdb_output))
-            return_status = False
-        log("Removing {}".format(corefile_fqpn))
-        os.unlink(corefile_fqpn)
+        if "core files" in summary:
+            # Process the core files
+            return_code |= self._process_core_files(os.path.split(destination)[0])
 
-    return return_status
+        return return_code
 
+    def _list_files(self, hosts, source, pattern, depth):
+        """List the files in source with that match the pattern.
 
-def get_test_category(test_file):
-    """Get a category for the specified test using its path and name.
+        Args:
+            hosts (NodSet): hosts on which the files are located
+            source (str): where the files are currently located
+            pattern (str): pattern used to limit which files are processed
+            depth (int): max depth for find command
 
-    Args:
-        test_file (str): the test python file
+        Returns:
+            tuple: a tuple containing:
+                int: 0 = success, 16 = failure
+                NodeSet: hosts with at least one file matching the pattern in the source directory
 
-    Returns:
-        str: concatenation of the test path and base filename joined by dashes
+        """
+        status = 0
+        hosts_with_files = NodeSet()
+        source_files = os.path.join(source, pattern)
+        logger.debug("-" * 80)
+        logger.debug("Listing any %s files on %s", source_files, hosts)
+        other = ["-printf", "'%M %n %-12u %-12g %12k %t %p\n'"]
+        result = run_remote(logger, hosts, find_command(source, pattern, depth, other))
+        if not result.passed:
+            message = f"Error determining if {source_files} files exist on {hosts}"
+            self._fail_test(self.result.tests[-1], "Process", message)
+            status = 16
+        else:
+            for data in result.output:
+                for line in data.stdout:
+                    if source in line:
+                        logger.debug("Found at least one file match on %s: %s", data.hosts, line)
+                        hosts_with_files.add(data.hosts)
+                        break
+                if re.findall(fr"{FAILURE_TRIGGER}", "\n".join(data.stdout)):
+                    # If a file is found containing the FAILURE_TRIGGER then report a failure.
+                    # This feature is used by avocado tests to verify that launch.py reports
+                    # errors correctly in CI. See test_launch_failures in harness/basic.py for
+                    # more details.
+                    logger.debug(
+                        "Found a file matching the '%s' failure trigger on %s",
+                        FAILURE_TRIGGER, data.hosts)
+                    message = f"Error trigger failure file found in {source} (error handling test)"
+                    self._fail_test(self.result.tests[-1], "Process", message)
```

Comment:
(pylint-no-value-for-parameter) No value for argument 'fail_reason' in staticmethod call